packages/router/router.ts

```diff
@@ -2038,14 +2035,13 @@ export function unstable_createStaticHandler(
   ): Promise<Omit<StaticHandlerContext, "location"> | Response> {
     let result: DataResult;
     if (!actionMatch.route.action) {
-      let href = createServerHref(new URL(request.url));
```

Comment:
This `createServerHref` utility came over from Remix and was used in a few places that it didn't make sense.  It's main purpose is to strip the hash from a URL because that doesn't get sent to the server, so really we want it in 2 places - (1) creating a submission and (2) creating a `Request` for a loader/action.  So I removed the usages of `createServerHref` and instead use a more targeted `stripHasFromPath` utility in those 2 places.